Scripts/RemediationScripts/Remediate-ConfigureConditionalAccessPolicyForPIM.ps1

```diff
@@ -285,19 +288,44 @@ function Configure-ConditionalAccessPolicyForPIM
     if ([String]::IsNullOrWhiteSpace($FilePath))
     {
         $policyAssignments = Get-AzRoleManagementPolicyAssignment -Scope "subscriptions/$($SubscriptionId)"
-
+        
+        # Fetch all the role definitions (built-in and custom).
+        $roleDefinitions = Get-AzRoleDefinition | Select-Object -ExpandProperty Name
+        
         if (-not $RoleName)
-        {
-            $adminRoles = @("Owner", "Contributor", "User Access Administrator")
+        {   
+            # These are the roles which are excluded by the control.
+            $excludedRoles = @(
+                "Azure Front Door Domain Contributor",
+                "Azure Front Door Domain Reader",
+                "Azure Front Door Profile Reader",
+                "Azure Front Door Secret Contributor",
+                "Azure Front Door Secret Reader",
+                "Defender for Storage Data Scanner",
+                "AzureML Compute Operator",
+                "Cognitive Services Usages Reader",
+                "Key Vault Crypto Service Release User"
+            )
+
+            # Remove excluded controls from all role definition.
+            $criticalRoles = $roleDefinitions | Where-Object { $_ -notin $excludedRoles }
```

Comment:
NIT: Better to use roleIds as name might change which is very less likely but still if we can do better.

Reply:
Adding it as a known issue in the docs.